utils/comparable_bytes.cc

```diff
@@ -230,6 +230,38 @@ static uint64_t timeuuid_msb_from_comparable_bytes(uint64_t byte_comparable_msb)
             | (byte_comparable_msb >> 48);
 }
 
+// Returns the number of significant digits in the big decimal.
+// For example, the "precision" of 12.34e56 is 4.
+std::size_t count_digits(const boost::multiprecision::cpp_int& value) {
+    // special case 0
+    if (value.is_zero()) {
+        return 1;
+    }
+
+    // Count the number of bits in the value.
+    // The value is stored as array of 'limbs' in boost::multiprecision::cpp_int
+    // and all limbs except the most significant limb will have bits_per_limb.
+    // So, total bits = (num of limbs - 1) * bits per limb + bits in ms limb.
+    const auto& backend = value.backend();
+    const auto limb_count = backend.size();
```

Comment:
Do we really need to access the internals?

Reply:
We can consider switching to boost::multiprecision::cpp_dec_float types if it makes it easier.